__TESTS__/unit/urlConfig.test.ts

```diff
@@ -0,0 +1,74 @@
+import TransformableImage from "../../src/transformation/TransformableImage";
+import Resize from "../../src/actions/resize/Resize";
+import ICloudinaryConfigurations from "../../src/interfaces/Config/ICloudinaryConfigurations";
+
+const DEMO_CONFIG = {
+  cloud: {
+    cloudName: 'demo'
+    // Other cloud-based properties
+  }
+};
+
+describe('It tests a combination of Cloudianry URL and Configuration', () => {
+  it ('Generates a URL', () => {
+    const url = new TransformableImage('my_image')
+      .setConfig(DEMO_CONFIG)
+      .toURL();
+
+    expect(url).toBe('http://res.cloudinary.com/demo/image/upload/my_image');
+  });
+  it ('Generates a URL with transforamtions', () => {
+    const url = new TransformableImage()
+      .setConfig(DEMO_CONFIG)
+      .resize(Resize.fill(100, 100))
+      .setPublicID('sample')
+      .describeAsset({
+        // has sane defaults for image/upload
+      })
+      .toURL();
+
+    expect(url).toBe('http://res.cloudinary.com/demo/image/upload/c_fill,w_100,h_100/sample');
+  });
+
+  it ('Shows a use-case for global configuration', () => {
+    //
+    /**
+     * We can implement this "wrapper", or instruct our customers how to implement it.
+     */
+    class MyGlobalCloudinary {
+      public cloudinaryConfig: ICloudinaryConfigurations;
+      // Constructor accepts a cloudinary configuration
+      constructor(cloudinaryConfig: ICloudinaryConfigurations) {
+        this.cloudinaryConfig = cloudinaryConfig;
+      }
+
+      // Utility function that returns a new instance with configuration set
+      image(publicID: string) {
+        return new TransformableImage(publicID)
+          .setConfig(this.cloudinaryConfig);
+      }
+    }
+
+    /**
+     * Define the configuration and your "global" instance
+     */
+    const myInstance = new MyGlobalCloudinary({
+      cloud: {
+        cloudName: 'MY_CLOUD_NAME'
+      }
+    });
+
+    /**
+     * Use the "global" instance
+     */
+    const url = myInstance.image('sample')
+      .resize(Resize.fill(100, 100))
+      .describeAsset({ // In PHP, thi
```

Comment:
Comment is cut

Reply:
Also, why not `assetDescriptor`? not sure `describeAsset` means much

Reply:
I think describeAsset means much more than assetDescriptor. (First of all, one is verb and the other is a noun, but also assetDescriptor is an internal term for Cloudinary, while describeAsset could be more generic.

No doubt that "describeAsset" is not the best name,  but assetDescriptor is exactly the same only with reversed words. 

I do want to find a better name though.